extensions/Webhooks/Extension.pm

```diff
@@ -143,10 +146,26 @@ sub user_preferences {
         delete_backlog_queue($webhook->id);
         $webhook->remove_from_db();
       }
+      $push->set_config_last_modified();
       $dbh->bz_commit_transaction;
 
-      update_push_connectors();
+    }else{
+
+      # save change(s)
 
+      my $dbh      = Bugzilla->dbh;
+      my $webhooks = Bugzilla::Extension::Webhooks::Webhook->match(
+        {user_id => $user->id});
+
+      $dbh->bz_start_transaction();
+      foreach my $webhook (@$webhooks) {
+        my $connector = $push->connectors->by_name('Webhook_' . $webhook->id);
+        my $config = $connector->config;
+        $config->{enabled} = trim($input->{$connector->name . ".enabled"});
```

Comment:
We may need to validate the input here to make sure that the value is either 'Enabled' or 'Disabled'. Sneaky users can modify values before POST to circumvent checks to do bad things.

Reply:
I hadn't thought about sneaky users before :open_mouth: